src/CacheTower/CacheStack.cs

```diff
@@ -293,12 +293,12 @@ private async ValueTask BackPopulateCacheAsync<T>(int fromIndexExclusive, string
 							CacheEntryStatus.Miss => CacheUpdateType.AddEntry,
 							_ => CacheUpdateType.AddOrUpdateEntry
 						};
-						await InternalSetAsync(cacheKey, refreshedEntry, cacheUpdateType);
+						await stack.InternalSetAsync(cacheKey, refreshedEntry, cacheUpdateType);
 
-						KeyLock.ReleaseLock(cacheKey, refreshedEntry);
+						stack.KeyLock.ReleaseLock(cacheKey, refreshedEntry);
 
 						return refreshedEntry;
-					}, (previousEntry, asyncValueFactory, settings, entryStatus, cacheKey), settings);
+					}, (previousEntry, asyncValueFactory, settings, entryStatus, cacheKey, this), settings);
```

Comment:
Passing the settings could be avoided if you would change the `Func<TState, ValueTask<CacheEntry<TValue>>>` to `unc<CacheSettings, TState, ValueTask<CacheEntry<TValue>>>` since the settings always need to be passed to `WithRefreshAsync`. I did not do this yet because it would require to introduce a breaking change on the interface which I wanted to avoid for now.